backend/src/apiserver/resource/resource_manager.go

```diff
@@ -179,21 +194,37 @@ func (r *ResourceManager) CreatePipeline(name string, description string, pipeli
 	if err != nil {
 		return nil, util.Wrap(err, "Create pipeline failed")
 	}
+	newPipeline.DefaultVersion.Status = model.PipelineVersionReady
+	err = r.pipelineStore.UpdatePipelineVersionStatus(
+		newPipeline.DefaultVersionId, newPipeline.DefaultVersion.Status)
+	if err != nil {
+		return nil, util.Wrap(err, "Create pipeline version failed")
+	}
 	return newPipeline, nil
 }
 
 func (r *ResourceManager) UpdatePipelineStatus(pipelineId string, status model.PipelineStatus) error {
 	return r.pipelineStore.UpdatePipelineStatus(pipelineId, status)
 }
 
+func (r *ResourceManager) UpdatePipelineVersionStatus(pipelineId string, status model.PipelineVersionStatus) error {
+	return r.pipelineStore.UpdatePipelineVersionStatus(pipelineId, status)
+}
+
 func (r *ResourceManager) GetPipelineTemplate(pipelineId string) ([]byte, error) {
 	// Verify pipeline exist
-	_, err := r.pipelineStore.GetPipeline(pipelineId)
+	pipeline, err := r.pipelineStore.GetPipeline(pipelineId)
 	if err != nil {
 		return nil, util.Wrap(err, "Get pipeline template failed")
 	}
 
-	template, err := r.objectStore.GetFile(storage.CreatePipelinePath(fmt.Sprint(pipelineId)))
+	var UUID string
+	if pipeline.DefaultVersion == nil {
+		UUID = pipeline.UUID
```

Comment:
this is not needed for now since all pipeline have default version 
and not needed in the future, because if DV is nil it should not return template.

Reply:
might just return a not found exception.

Reply:
Done. (Note: originally meant for this to handle the case if the databack fill is not successful. E.g., default version id in pipelines table is not set. )